lib/dogwatch/model/options.rb

```diff
@@ -55,6 +64,24 @@ def notify_audit(notify = false)
       def include_tags(include = true)
         @attributes.include_tags = !!include
       end
+
+      # @param [Hash{String=>Fixnum}] thresholds
+      def thresholds(thresholds)
+        validate_monitor_type_specific_option!(:thresholds)
+        @attributes.thresholds = thresholds
+      end
+
+      private
+
+      def validate_monitor_type_specific_option!(option)
+        options = Array(MONITOR_TYPE_OPTIONS_MAP[@monitor_type])
+        return true if options.include?(option)
+
+        # rubocop:disable Metrics/LineLength
+        message = "The #{@monitor_type.inspect} monitor type does not support #{option.inspect}."
+        message << " Did you mean one of #{options.join(', ')}?" if options.any?
+        raise NotImplementedError, message
```

Comment:
Is there anything that actually enforces that the user sets a monitor type (I haven't used Dogwatch so forgive me if it's required).  If `monitor_type` can be nil, should we have a error message saying you should set a type?